lib/utils.js

```diff
@@ -205,6 +205,31 @@ utils.getResourceUID = function getResourceUID(namespace, resource) {
         .update(namespace + resource, 'binary').digest('hex');
 };
 
+/**
+ * Validate headers per testing if readable
+ * @param {object} headers - headers attached to the http request (lowercased)
+ * @return {boolean} - returns true if headers are readable, false if not
+ */
+utils.checkReadableHeaders = function checkReadableHeaders(headers) {
+    // check every header except the x-amz-meta- ones for AWS compatibility
+    const isReadableHeaders = Object.keys(headers).filter(h =>
+    !h.startsWith('x-amz-meta-')).every(headerKey => {
+        const headerVal = String(headers[headerKey]);
+        let isReadableChar = true;
+        // check that each character is readable
+        for (let i = 0; i < headerVal.length; i++) {
+            const ch = headerVal.charCodeAt(i);
+            if (ch >= 32 && ch <= 127) {
```

Comment:
Could this be reduced to a single `if` condition instead of an `if...else` statement?